docs/standard/exceptions/best-practices-for-handling-and-using-exceptions-outline.md

```diff
@@ -0,0 +1,91 @@
+			1.	Identify the specific exception you want to handle.
+			2.	Make sure you can restore your application to a known state.
+			3.	Identify the point from which you want the process to restart.
+
+2.	Guidelines for using try/catch blocks.
+	1.	Use a global exception handler.
+		i.	Use one or more of the .net event handlers.
+			1.	DispatcherUnhandledException
+			2.	AppDomain.CurrentDomain.UnhandledException
+			3.	System.Threading.Tasks.TaskScheduler.UnobservedTaskException 
+			4.	Dispatcher.UnhandledException
+		ii.	Execute your main method in a try/catch block.
+		iii. Fewer try/catch blocks are preferential to more.  Many small applications need only the global exception handlers.
+		iv.	A global handler in a library is usually unnecessary.  Libraries should let unhandled exceptions flow up to the global handler.
+	2.	Use try/catch blocks to isolate code that may throw exceptions for reasons beyond your control.
+		i.	When isolating a section of code within a try/catch block put as little code as possible in the try block.  A single line is often sufficient.
+		ii.	Do not put code in a try block unless you are able to handle one or more of the exceptions that code may raise in the corresponding catch block. 
+		iii.	Prefer to catch specific exceptions that you are prepared to handle. If code throws a general exception or a type of exception that you do not expect consider letting the exception flow up to the global exception handler.
+		iv.	A try/catch block is not a substitute for proactively preventing exceptions. 
+	3.	Investigate thoroughly before catching Exception.  
+		i.	Most catch blocks should handle specific exceptions only i.e. IOExceptions.  
+		ii.	Prefer to let the global exception handler catch Exception.
+	4.	Only catch exceptions you actually handle.
+		i.	Logging an exception is not handling it. Logging belongs in the global exception handler.
+		ii.	Catching an exception, collecting diagnostic info, and re-throwing is a good practice.
+			1.	If you create a new exception object be sure to set the InnerException property to the original exception.  This ensures you preserve the stack trace.
+		iii.	Properly handling an exception often involves recovery and/or retry.  Your handling logic should be able to perform these tasks if required.
+			1.	Assess the specific type of error you are handling.
+			2.	Verify that your environment is in a known state. 
+			3.	Attempt to cancel or rollback any data transactions.
+		iv.	Handle only the code inside the try block.  If necessary, re-throw the error and let higher level code clean up as required.
+	5.	Shutting the application down or returning to the main method or home page after an exception is often the best choice.
+		i.	An investment in preventing exceptions is preferred to an investment in handling them.
+	6.	Use a finally block to execute code that should run even if an error occurs.		

```

Comment:
This should be given more visibility: `try`/`finally` is necessary when resource cleanup is needed, and a `using` statement doesn't suffice.